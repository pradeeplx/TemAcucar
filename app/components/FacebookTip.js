import React from 'react'
import { fontFactor } from "../helpers"
import Colors from "../Colors"
import Tip from "./Tip"
import Sentence from "./Sentence"
import FacebookButton from "./FacebookButton"

export default FacebookTip = ({ onPress, loading, title }) => (
  <Tip>
    <Sentence style={{
      color: Colors.black,
      fontSize: 10 * fontFactor(),
      marginBottom: 10,
      textAlign: 'center',
      marginHorizontal: 40,
    }}>
      { title || 'Quer ter acesso a funcionalidades incr√≠veis?' }
    </Sentence>
    <FacebookButton
      onPress={onPress}
      isLoading={loading}
      style={{
        width: 220 * fontFactor(),
      }}
    >
      Conecte sua conta do Facebook
    </FacebookButton>
  </Tip>
)
