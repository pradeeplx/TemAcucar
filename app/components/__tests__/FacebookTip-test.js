import { View } from 'react-native'
import React from 'react'
import FacebookTip from '../FacebookTip'
import renderer from 'react-test-renderer'

it('renders correctly', () => {
  const tree = renderer.create(
    <FacebookTip title="Foo" />
  ).toJSON();
  expect(tree).toMatchSnapshot()
});

it('renders default title', () => {
  const tree = renderer.create(
    <FacebookTip />
  ).toJSON();
  expect(tree).toMatchSnapshot()
});
