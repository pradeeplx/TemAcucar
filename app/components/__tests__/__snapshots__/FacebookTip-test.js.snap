exports[`test renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "borderColor": "#FAF6F1",
        "borderTopWidth": 12,
        "padding": 15,
      },
      undefined,
    ]
  }>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#654C50",
          "fontFamily": "Avenir",
          "fontSize": 18,
        },
        Object {
          "color": "#000000",
          "fontSize": 15,
          "marginBottom": 10,
          "marginHorizontal": 40,
          "textAlign": "center",
        },
      ]
    }>
    Foo
  </Text>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "#557BD3",
        "borderColor": "#557BD3",
        "borderRadius": 27,
        "borderWidth": 0.5,
        "flexDirection": "row",
        "height": 54,
        "justifyContent": "center",
        "marginBottom": 0,
        "opacity": 1,
        "paddingHorizontal": 30,
        "paddingVertical": 0,
        "width": 330,
      }
    }
    testID={undefined}>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "flex": 1,
            "fontSize": 18,
            "textAlign": "center",
          },
          Array [
            Object {
              "color": "#FF7F85",
              "fontFamily": "Avenir",
              "fontSize": 18,
              "textAlign": "center",
            },
            Array [
              Object {
                "color": "#FFFFFF",
              },
              undefined,
            ],
          ],
        ]
      }>
      Conecte sua conta do Facebook
    </Text>
  </View>
</View>
`;

exports[`test renders default title 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "borderColor": "#FAF6F1",
        "borderTopWidth": 12,
        "padding": 15,
      },
      undefined,
    ]
  }>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#654C50",
          "fontFamily": "Avenir",
          "fontSize": 18,
        },
        Object {
          "color": "#000000",
          "fontSize": 15,
          "marginBottom": 10,
          "marginHorizontal": 40,
          "textAlign": "center",
        },
      ]
    }>
    Quer ter acesso a funcionalidades incr√≠veis?
  </Text>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "#557BD3",
        "borderColor": "#557BD3",
        "borderRadius": 27,
        "borderWidth": 0.5,
        "flexDirection": "row",
        "height": 54,
        "justifyContent": "center",
        "marginBottom": 0,
        "opacity": 1,
        "paddingHorizontal": 30,
        "paddingVertical": 0,
        "width": 330,
      }
    }
    testID={undefined}>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "flex": 1,
            "fontSize": 18,
            "textAlign": "center",
          },
          Array [
            Object {
              "color": "#FF7F85",
              "fontFamily": "Avenir",
              "fontSize": 18,
              "textAlign": "center",
            },
            Array [
              Object {
                "color": "#FFFFFF",
              },
              undefined,
            ],
          ],
        ]
      }>
      Conecte sua conta do Facebook
    </Text>
  </View>
</View>
`;
